scripts/build/gen_relocate_app.py

```diff
@@ -529,11 +536,16 @@ def create_dict_wrt_mem():
     input_rel_dict = args.input_rel_dict.read()
     if input_rel_dict == '':
         sys.exit("Disable CONFIG_CODE_DATA_RELOCATION if no file needs relocation")
+
+    # Replace pipes used in symbol filter regular expression
+    input_rel_dict = input_rel_dict.replace("\\|", "##PIPE##")
```

Comment:
I know the `|` is used widely in regex'es, so it's natural to support them directly, but going through this substitution seems less than optimal to me.

This PR https://github.com/zephyrproject-rtos/zephyr/pull/60999 changed the input relocation argument from being a string to be a file.

Being a file actually means that there is no reason for a single string because a file can be multi-line.
I suggest to make a #60999 follow-up commit which remove the use of `|` as separator and instead create a multi-line input file.

Doing so will free the use of `|` and thereby remove the need for special handling in the regex.